### DISCUSSION TITLE
How do you use the SDK from node?

### DISCUSSION THREAD

**Discussion by ollema** [pocketbase/pocketbase#272] [Q&A] [answered]
Hi!

Thanks for a super cool project!

I'm trying to programmatically create a "migration" or "seed" script for the collections, since I dislike having config not stored in version control.

I have the following script:
`scripts/seed_database.js`
```js
import PocketBase from 'pocketbase';

const pocketBaseUrl = 'http://localhost:8090';

console.log(`will seed database at ${pocketBaseUrl}`);

const client = new PocketBase(pocketBaseUrl);
const user = process.env.POCKET_BASE_ADMIN_USER;
const password = process.env.POCKET_BASE_ADMIN_PASS;

if (user === undefined || password === undefined) {
	throw Error('POCKET_BASE_ADMIN_USER or POCKET_BASE_ADMIN_PASS not set');
}

console.log(`${user}:${password}`);

await client.admins.authViaEmail(user, password);
```
when I run `node scripts/seed_database.js` I get:
```
...
ClientResponseError 0: Something went wrong while processing your request.
    at new i (file:///home/s0001325/work/sersophane/node_modules/.pnpm/pocketbase@0.3.1/node_modules/pocketbase/dist/pocketbase.es.mjs:1:2078)
    at file:///home/s0001325/work/sersophane/node_modules/.pnpm/pocketbase@0.3.1/node_modules/pocketbase/dist/pocketbase.es.mjs:1:25282
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async file:///home/s0001325/work/sersophane/scripts/seed_database.js:17:1 {
  url: '',
  status: 0,
  data: {},
  isAbort: false,
  originalError: null
}

Node.js v18.7.0
```
Environment variables are set correctly

Is this expected? Am I doing something weird here? I am not very experienced with js

The idea is to authenticate and then to call a bunch of `client.collections.create` so I don't have to click around in the UI and try to remember every field setting whenever I want to redeploy.

Thanks in advance, let me know if I should create an issue instead.

**Comment by ganigeorgiev** [comment, marked as answer]
This looks similar to https://github.com/pocketbase/pocketbase/issues/258. Try changing the SDK backend url to an ipv4, eg:
```js
const pocketBaseUrl = 'http://127.0.0.1:8090';
```
The reason this could resolve the issue is that go dns resolver (and most likely in other languages too) by default will first try to resolve to IPv4 then IPv6 addresses, while in Node17+ for some reason this is the opposite - IPv6 first, then IPv4. I'll try to improve the SDK error reporting in future versions.

Let me know if the above doesn't work for you and I'll investigate it further.

_btw. The next v0.4.0 release (sometime next week) will add support for export/import of a collection schema via an API endpoint and the Admin UI. Additionally, when PocketBase is used as framework, a console command will be also available to generate a migration file with your current local collections configuration._

**Reply by ganigeorgiev** [reply]
I've opened an issue in the JavaScript SDK repo for the error reporting - https://github.com/pocketbase/js-sdk/issues/21

**Reply by ollema** [reply]
```js
const pocketBaseUrl = 'http://127.0.0.1:8090';
```
this works, thanks!

---

the feature you describe in the upcoming release sounds great! then I can discard this script
